src/targets/ref/CMakeLists.txt

```diff
@@ -27,14 +27,14 @@ add_library(migraphx_ref
     lowering.cpp
     gemm.cpp
 )
+target_link_options(migraphx_ref PUBLIC "-Wl,--no-as-needed")
```

Comment:
We shouldn't add these flags to the link options since this can affect how things are linked after the target.